### DISCUSSION TITLE
Remote sensor installation issues

### DISCUSSION THREAD

**Discussion by Hammer-Bear** [Security-Onion-Solutions/securityonion#6165] [Unsupported Versions] [answered]
Hi,

Im currently running through installing a remote sensor in a distributed environment and it runs through the install until it hits trying to authenticate with salt, it took the soremote creditials when prompted and after some time just errored out. I had a look at the `sosetup.log` and it appears to be timing out when trying to auth with salt `"Attempt to authenticate with the salt master failed with timeout error"` not sure where else to go as we've already installed one remote sensor and that worked absolutely fine but this one appears to be having issues. Currently running SO 2.3.61

**Comment by Rbartram** [comment]
Hi Dan.

I'm not sure it's the exact same issue, but I just battled a similar issue installing a sensor node using 2.3.80 iso.  It turned out that following line was missing from my /etc/sudoers file.

    soremote ALL=(ALL) NOPASSWD:/usr/bin/salt-key

Without this line in to the sudoers file, the sensors salt-key cannot be remotely accepted during the sensor setup.

**Comment by TOoSmOotH** [comment]
Most likely you have a firewall in-between blocking the connection.

**Reply by Hammer-Bear** [reply]
I've added all the ports recommended in https://docs.securityonion.net/en/2.3/firewall.html however it's still failing to install. When looking in the `/root/sosetup.log` I get the following

```
#########################################
#########################################
###                                                                           ###
###     UNAUTHORISZED ACCESS PROHIBITED       ###
###                                                                           ###
#########################################
#########################################
Adding (Sensor name)
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Attempt to authenticate with the salt master failed with timeout error
Finishing up network setup
... Verifying all network devices are managed by Network Manager
... Copying 99-so-checksum-offload-disable
...  Modifying 99-so-checksum-offload-disable
/home/(admin)/SecuirtyOnion/setup/../salt/common/tools/sbin/so-ssh-harden: line 3: /usr/sbin/so-common: no such file or directory
Copying current config to /tmp/sshd_config
Reloading sshd to load config changes
[ WARNING ] Any new ssh sessions will need to remove and reaccept the host key fingerprints for this server before reconnecting.
Enabling checkin at boot 
Attempt to authenticate with the salt master failed with timeout error
Errors detected during setup: skipping post-setup steps to allow for analysis of failures
2021-11-23T09:15:38Z | I | Removing so-setup permission entry from sudoers file
```
to me it looks like it hangs a little but then establishes a connection but still causes it to fail?

**Reply by Rbartram** [reply]
Dan, try running `sudo salt-key` on your manager node to check the state of the salt key for the sensor node.   In what state is the key for the sensor?

**Reply by Hammer-Bear** [reply]
so I've been using this as well to see if it even hits salt and nothing appears in any of the lists which is whats confusing me because I cant work out if it is or isn't connecting to salt master

![image](https://user-images.githubusercontent.com/36236670/143059255-46e025e4-d8e2-4b85-8183-26f0d7fdce69.png)

**Comment by m0duspwnens** [comment]
Can you confirm that the remote node is able to connect to the manager via ssh with the soremote user?

**Reply by Hammer-Bear** [reply]
Yes can SSH to the manager from the sensor via soremote, iv'e also noticed that its added the IP to the firewall groups under minions and also shows in /opt/so/saltstack/local/pillar/data/sensortab.sls under the other sensors

**Comment by Hammer-Bear** [comment, marked as answer]
Ok found the issue after watching the firewalls on both sides, there was some weird routing once it hit the remote site so it could communicate with the manager but the manager to sensor just got lost. built the sensor on another remote location and works fine.

*EDIT*
Ok so it gets past the original issue we had but now its not showing in grid or grafana but is showing in `sudo salt-key`, `opt/so/saltstack/local/pillar/data/sensorstab.sls` and also the firewall groups under minion

**Reply by Hammer-Bear** [reply]
found the issue, networks team set up the nat rules incorrectly on the firewall. now corrected and is working as should be
